recipes/eooffshore_ics_cms_WIND_GLO_WIND_L3_OBSERVATIONS_Metop_ASCAT/recipe.py

```diff
@@ -0,0 +1,259 @@
+                if not missing_product(satellite, date, satellite_pass)
+            ]
+            self._create_file_pattern(file_list=file_list)
+
+        self.process_input = lambda ds, fname: process_ds(
+            ds=ds,
+            fname=fname,
+            satellites=self.satellite_dates.keys(),
+            service=self.service,
+        )
+        super().__post_init__()
+
+        # Insert additional stage into pipeline that sorts the
+        # ASCAT pattern products according to measurement_time (spatial mean)
+        self._original_stages = self._compiler().stages
+        self._compiler = self._ascat_xarray_zarr_recipe_compiler
+
+    def _ascat_xarray_zarr_recipe_compiler(self) -> Pipeline:
+        """Custom pipeline compiler containing additional sorting stage"""
+        stages = list(self._original_stages)
+        stages.insert(
+            1, Stage(name="sort_file_pattern", function=self._sort_file_pattern)
+        )
+        return Pipeline(stages=stages, config=self)
+
+    def _create_file_pattern(self, file_list: List[str]) -> None:
+        """Creates the recipe file pattern from the specified list"""
+        self.file_pattern = pattern_from_file_sequence(
+            file_list=file_list,
+            concat_dim="time",
+            nitems_per_file=1,
+            # Must be set to unknown, as default type of 'netcdf4' will prevent xarray engine from being specified
+            file_type="unknown",
+            fsspec_open_kwargs={
+                "username": "rabernathey",
+                "password": "oCg!25s%DN^M",
+                "block_size": 400_000_000,
+            },
+        )
+
+    def _sort_file_pattern(self, *, config: XarrayZarrRecipe) -> None:
+        """Replace the recipe file pattern with new pattern containing time-sorted file URLs"""
+        file_times = []
+        for input_key, input_filename in self.file_pattern.items():
+            with open_input(input_key=input_key, config=self) as ds:
+                # Exclude files with NaT measurement_time in the source data product
+                if ds.dims["time"] == 1:
+                    file_times.append((ds.time.values[0], input_filename))
+        # New pattern with files sorted according to (measurement_)time
+        self._create_file_pattern(
+            file_list=[input_filename for _, input_filename in sorted(file_times)]
+        )
+
+    def _make_url(self, satellite: str, date: pd.Timestamp, satellite_pass: str) -> str:
+        """Creates an ASCAT product FTP URL"""
+        year = date.strftime("%Y")
+        month = date.strftime("%m")
+        day = date.strftime("%d")
+        nrt = self.host.startswith("nrt")
+        return f"ftp://{self.host}/Core/{self.service}/KNMI-GLO-WIND_L3-{'REP-' if not nrt else ''}OBS_METOP-{satellite}_ASCAT_12_{satellite_pass}{'_V2' if nrt else ''}/{year}/{month}/GLO-WIND_L3-OBS_METOP-{satellite}_ASCAT_12_{satellite_pass}_{year}{month}{day}.nc"
+
+
+setup_logging()
```

Comment:
```suggestion
```

Reply:
Thanks @cisaacstern, I've removed the `setup_logging` references in the latest commit.